### ISSUE TITLE
JavaScript syntax highlighting detects wrong string start in regex

### ISSUE THREAD

**Issue by jublo** [Microsoft/vscode#248]
- Visual Studio Code 0.10.0
- OS X 10.11.1

```javascript
    var _url = function (data) {
        if ((/boolean|number|string/).test(typeof data)) {
            return encodeURIComponent(data).replace(/!/g, "%21").replace(/'/g, "%27").replace(/\(/g, "%28").replace(/\)/g, "%29").replace(/\*/g, "%2A");
        } else {
            return "";
        }
    };
```

![bildschirmfoto 2015-11-19 um 16 05 17](https://cloud.githubusercontent.com/assets/6348321/11274280/28f1f0f6-8ed8-11e5-9e2c-6e22dd7e5f55.png)

**Comment by BlackFrog1** [comment]
This has been fixed in version 0.10.1 
<img width="839" alt="screen shot 2015-11-19 at 13 50 18" src="https://cloud.githubusercontent.com/assets/550820/11280600/bf6be3f2-8ec4-11e5-9662-4d07850a37ac.png">

I am also on OS X 10.11.1

**Comment by jublo** [comment]
@BlackFrog1 Thanks, confirmed!
